src/components/Datatable/ControlsModule/ColumnsControls/hooks/useColumnOrder.ts

```diff
@@ -0,0 +1,77 @@
+import { useEffect, useState } from 'react';
+import { comparator, keys, lt, sort } from 'ramda';
+import deepEqual from 'fast-deep-equal';
+
+import { DatatableStore } from '../../../Datatable.store';
+
+const compare = (order) => (a, b) => {
```

Comment:
an explanation of what the logic is meant to achieve here and a unit test would be helpful.